pkg/controllers/secretstore/client_manager.go

```diff
@@ -257,3 +265,16 @@ func assertStoreIsUsable(store esv1beta1.GenericStore) error {
 	}
 	return nil
 }
+
+func matchStringInList(list []string, item string) (bool, error) {
```

Comment:
`list` is a bit too vague in this context. Could you rename the variable `regexes`? Also `item` to `namespace`. I would name the function `matchNamespaceRegexes` or just write the logic in the shouldProcessSecret method directly.